core/main.go

```diff
@@ -453,25 +457,18 @@ func (a *AutoSpotting) handleNewSpotInstanceLaunch(r *region, i *instance) error
 		return fmt.Errorf("region %s is missing asg data", i.region.name)
 	}
 
+	if len(a.config.sqsReceiptHandle) == 0 {
+		if asg.isEnabledForEventBasedInstanceReplacement() {
+			i.region.sqsSendMessageSpotInstanceLaunch(asgName, i.InstanceId, i.State.Name)
+		}
+		return nil
+	}
+	defer i.region.sqsDeleteMessage(i.InstanceId)
+
 	logger.Printf("%s Found instance %s is not yet attached to its ASG, "+
 		"attempting to swap it against a running on-demand instance",
 		i.region.name, *i.InstanceId)
 
-	hasHooks, err := asg.hasLaunchLifecycleHooks()
```

Comment:
so we no longer handle these launch lifecycle hooks?

Reply:
we handle it by sending a message to the sqsqueue.
But we no more need to check in spot event start if ASG has LifecycleHooks as LifecycleHooks is used when we do not have events enabled